### PR TITLE
chore: Update op-geth to latest version

### PR DESCRIPTION
**Pull request by mslipper**
Update op-geth to the latest version on the `optimism-history` branch. This version exposes the customizable simulator backend that we need to merge https://github.com/ethereum-optimism/optimism/pull/3322.

Also adds a small script to automatically update the op-geth version in the future.

### CODE REVIEW

File: op-chain-ops/go.mod [hunk 1]
```diff
@@ -58,4 +58,7 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-replace github.com/ethereum/go-ethereum v1.10.21 => github.com/ethereum-optimism/op-geth v0.0.0-20220904174542-4311f9d2cead
+replace (
+	github.com/ethereum/go-ethereum v1.10.21 => github.com/ethereum-optimism/op-geth v0.0.0-20220904174542-4311f9d2cead
```
Comment (trianglesphere) [inline op-chain-ops/go.mod:62]: Why are we requiring two different versions of geth here?

**Review by trianglesphere — COMMENTED** [review]
I think there is one double import, but overall looking good. Having the script is super nice rather than doing it by hand.

File: op-chain-ops/go.mod (hunk 1 above)
Comment (mslipper) [reply op-chain-ops/go.mod:62]: Nice catch. We shouldn't have to different versions. Fixing.

**Comment by mslipper** [comment]
Tests will run once CircleCI fixes its outage.

File: ops/scripts/update-op-geth.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+export GOPRIVATE="github.com/ethereum-optimism"
+
+for value in $PROJECTS
+do
+	echo "Updating $value"
+	cd "./$value"
+	go mod edit -replace "github.com/ethereum/go-ethereum@$GETH_VERSION=github.com/ethereum-optimism/op-geth@optimism-history"
+	go mod tidy
+	cd ../
+done
```
Comment (protolambda) [inline ops/scripts/update-op-geth.sh:15]: Let's add a `go work edit -replace ...........`

**Review by protolambda — CHANGES_REQUESTED** [review]
`go.work` also contains a reference to op-geth. The script should include that too.

File: ops/scripts/update-op-geth.sh (hunk 2 above)
Comment (mslipper) [reply ops/scripts/update-op-geth.sh:15]: ah good call, always forget that.

**Review by protolambda — APPROVED** [review]

File: go.work.sum [hunk 3]
```diff
@@ -27,6 +27,8 @@ github.com/envoyproxy/go-control-plane v0.9.9-0.20210217033140-668b12f5399d/go.m
 github.com/envoyproxy/go-control-plane v0.9.10-0.20210907150352-cf90f659a021/go.mod h1:AFq3mo9L8Lqqiid3OhADV3RfLJnjiw63cSpi+fDTRC0=
 github.com/ethereum-optimism/op-geth v0.0.0-20220904174542-4311f9d2cead/go.mod h1:EYFyF19u3ezGLD4RqOkLq+ZCXzYbLoNDdZlMt7kyKFg=
 github.com/ethereum-optimism/op-geth v0.0.0-20220906163738-712cb0a1c140/go.mod h1:EYFyF19u3ezGLD4RqOkLq+ZCXzYbLoNDdZlMt7kyKFg=
```
Comment (tynes) [inline go.work.sum:29]: Are there supposed to be multiple versions of `op-geth` here?

File: ops/scripts/update-op-geth.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+def main():
+	for project in ('op-service', 'op-node', 'op-proposer', 'op-batcher', 'op-bindings', 'op-chain-ops', 'op-e2e'):
+		print(f'Updating {project}...')
+		update_mod(project)
+
+	# pull the replacer from one of the modules above, since
+	# go work edit -replace does not resolve the branch name
+	# into a pseudo-version.
+	pv = None
+	with open('./op-service/go.mod') as f:
```
Comment (tynes) [inline ops/scripts/update-op-geth.py:22]: This only needs to run in `op-service`?

File: op-chain-ops/deployer/deployer.go [hunk 5]
```diff
@@ -39,14 +38,17 @@ type Deployment struct {
 type Deployer func(*backends.SimulatedBackend, *bind.TransactOpts, Constructor) (common.Address, error)
 
 func NewBackend() *backends.SimulatedBackend {
-	return backends.NewSimulatedBackendWithCacheConfig(
```
Comment (trianglesphere) [inline op-chain-ops/deployer/deployer.go:42]: Is this a regression with deployments?

**Review by tynes — APPROVED** [review]

**Review by trianglesphere — APPROVED** [review]
How much of this tooling will we get to delete if we are using tags rather than commit hashes/branches?

File: op-chain-ops/deployer/deployer.go (hunk 5 above)
Comment (mslipper) [reply op-chain-ops/deployer/deployer.go:42]: Nope, the API was updated in `op-geth`.

**Comment by mslipper** [comment]
> How much of this tooling will we get to delete if we are using tags rather than commit hashes/branches?

The bit that resolves the branch to the pseudo-version in the updater script. We'll still need the rest.

**Pull request merged**